lib/inventory/controller.go

```diff
@@ -232,14 +417,18 @@ func (c *Controller) handlePingRequest(handle *upstreamHandle, req pingRequest)
 		}
 		return trace.Wrap(err)
 	}
-	handle.pings[handle.pingCounter] = pendingPing{
+	handle.pings[req.id] = pendingPing{
 		start: start,
 		rspC:  req.rspC,
 	}
 	return nil
 }
 
-func (c *Controller) handleHeartbeat(handle *upstreamHandle, hb proto.InventoryHeartbeat) error {
+func (c *Controller) handleHeartbeatMsg(handle *upstreamHandle, hb proto.InventoryHeartbeat) error {
+	// XXX: when adding new services to the heartbeat logic, make sure to also update the
```

Comment:
`XXX:` is an oldschool comment convention, kind of like `NOTE:` or `TODO:`.  I've always understood it to be basically equivalent to `NOTE:`, except that it conveys a higher severity, uncertainty, and/or exceptionalism.  Typically used to convey a warning that the reader should keep in mind when reading/modifying the associated code, or to convey a note about something the author is especially concerned or uncertain about.

Wikipedia's list of common comment tags describes `XXX` as being used to highlight problematic or misleading code.  I think the latter sense is more common. `FIXME` or `TODO` seem more appropriate for code that is just objectively problematic.

Its used in a few other places in the codebase... tho, looking at `git blame`, I think all the other places its used were also added by me lol.

I'm going to leave as-is unless I hear otherwise.  I think even if one isn't familiar with the tag, its achieves its primary purpose, which is that it is visually distinct and encourages the reader to take special note of it.